### PR TITLE
Adjustments to make it possible to execute the `EasyReflectometryApp`

### PR DESCRIPTION
**Pull request by andped10**
Main thing is to pass the list of `skip` strings to all nested `as_dict` methods.

### CODE REVIEW

File: tests/experiment/test_model.py [hunk 1]
```diff
@@ -391,11 +392,37 @@ def test_repr_resolution_function(self):
         )
 
 
+def test_copy():
```
Comment (rozyczko) [inline tests/experiment/test_model.py:395]: you didn't overwrite `copy` in class Model - why the need to test it here?

File: tests/experiment/test_model.py [hunk 2]
```diff
@@ -419,3 +446,18 @@ def test_dict_round_trip(interface):  # , additional_layer):
             model.interface().fit_func([0.3], model.unique_name),
             model_from_dict.interface().fit_func([0.3], model_from_dict.unique_name),
         )
+
+
+def test_dict_skip_unique_name():
```
Comment (rozyczko) [inline tests/experiment/test_model.py:451]: The `skip` argument is being tested elsewhere. Is there a reason to explicitly test this here?

**Review by rozyczko — APPROVED** [review]
The code looks fine. Minor questions about the tests added, but nothing prohibiting the merge.

File: tests/experiment/test_model.py (hunk 1 above)
Comment (andped10) [reply tests/experiment/test_model.py]: Moved to `test_topmost_nesting.py`.  These are test to ensure that the nested structure is functional by using a method from the topmost class

File: tests/experiment/test_model.py (hunk 2 above)
Comment (andped10) [reply tests/experiment/test_model.py]: Moved to `test_topmost_nesting.py`.  These are test to ensure that the nested structure is functional by using a method from the topmost class

**Pull request merged**